### PR TITLE
GAWB-986: Library attribute definitions

### PR DESCRIPTION
**Pull request by davidangb**
The bulk of this story is the src/main/resources/library/attribute-definitions.json file. This contains a first-draft/best-guess at aligning the attribute definitions from the two attachments to GAWB-916 (which don't always agree). Goal of this story is to create a draft, and allow edits later.

The format of that file is JSON Schema (http://json-schema.org/). By adhering to that standard, we inherit a bunch of syntax - we also enable the future ability to use attribute-definitions.json to validate the data we collect. Note that I've added a few custom fields on top of JSON Schema: "propertyOrder" to define how the order in which the UI renders fields, "inputHint" to provide end-user guidance in the UI. Everything else is standard. _TODO someday: create a JSON schema to validate these extensions._

The UI can read these definitions directly from orchestration, at /schemas/library-attributedefinitions-v1.

There's also a bit of refactoring in orchestration around reading resource files.

(This PR needs a rebase!)
- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: JIRA ticket checks:
  - Acceptance criteria exists and is met
  - Note any changes to implementation from the description
  - Add notes on what you've tested
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Update FISMA documentation if changes to:
  - Authentication
  - Authorization
  - Encryption
  - Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [x] Tell the tech lead (TL) that the PR exists if they wants to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
- Review cycle:
  - LR reviews
  - Rest of team may comment on PR at will
  - **LR assigns to submitter** for feedback fixes
  - Submitter rebases to develop again if necessary
  - Submitter makes further commits. DO NOT SQUASH
  - Submitter updates documentation as needed
  - Submitter **reassigns to LR** for further feedback
- [x] **TL** sign off
- [x] **LR** sign off
- [x] **Assign to submitter** to finalize
- [x] **Submitter**: Verify all tests go green, including CI tests
- [x] **Submitter**: Squash commits and merge to develop
- [x] **Submitter**: Delete branch after merge
- [x] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [x] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [x] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [x] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

**Comment by dvoet** [comment]
I see that we are providing "user friendly" metadata labels (e.g. Dataset Depositor for datasetDepositor). I am concerned about this because showing only these on the UI will lead to problems when people want to write method config expressions that use the values.  How will the user know they need to use datasetDepositor to reference the thing in the UI shown as "Dataset Depositor"?

**Comment by davidangb** [comment]
@dvoet I haven't heard any use cases yet for people writing expressions against this metadata ... that said, I don't see any reason why we CAN'T display both the friendly name and the key.

**Comment by dvoet** [comment]
@davidangb I am just raising the issue... I don't want to dictate solutions... this time.

File: src/main/scala/org/broadinstitute/dsde/firecloud/FireCloudServiceActor.scala [hunk 1]
```diff
@@ -98,8 +100,8 @@ class FireCloudServiceActor extends HttpServiceActor with FireCloudDirectives {
             serveIndex()
           } ~
             pathSuffix("api-docs") {
-              complete {
-                getResourceFileContents("swagger/api-docs.yaml")
+              withResourceFileContents("swagger/api-docs.yaml") { apiDocs =>
```
Comment (dmohs) [inline src/main/scala/org/broadinstitute/dsde/firecloud/FireCloudServiceActor.scala:103]: Nice.

**Comment by dmohs** [comment]
👍

**Pull request merged**